cep18/src/main.rs

```diff
@@ -425,39 +652,49 @@ pub fn install_contract(name: &str) {
         ENABLE_MINT_BURN.to_string(),
         storage::new_uref(enable_mint_burn).into(),
     );
+
     let entry_points = generate_entry_points();
 
-    let hash_key_name = format!("{HASH_KEY_NAME_PREFIX}{name}");
+    let mut message_topics = BTreeMap::new();
+    if [EventsMode::Native].contains(&events_mode.try_into().unwrap_or_default()) {
```

Comment:
```suggestion 
if events_mode.try_into().unwrap_or_default() == EventsMode::Native {
```

Please check if correct

Reply:
Thanks for the catch it actually wasn't needed an extra modality there